app/gui/view/component-browser/searcher-list-panel/src/column_grid.rs

```diff
@@ -0,0 +1,189 @@
+
+
+// =============
+// === Model ===
+// =============
+
+#[derive(Clone, Debug, CloneRef)]
+pub struct Model {
+    app:            Application,
+    display_object: display::object::Instance,
+    content:        Rc<RefCell<Vec<component_group::View>>>,
+    size:           Rc<Cell<Vector2>>,
+}
+
+impl Model {
+    fn new(app: &Application) -> Self {
+        let logger = Logger::new("ColumnGrid");
+        let app = app.clone_ref();
+        let display_object = display::object::Instance::new(&logger);
+
+        Self {
+            app,
+            display_object,
+            content: Rc::new(RefCell::new(vec![])),
+            size: Rc::new(Cell::new(Vector2::default())),
+        }
+    }
+
+    fn set_content(&self, content: &[LabeledAnyModelProvider]) -> Vector2 {
+        let overall_width = crate::WIDTH_INNER - 2.0 * crate::PADDING_INNER;
+        let column_width = (overall_width - 2.0 * COLUMN_GAP) / 3.0;
+        let content = content
+            .iter()
+            .map(|LabeledAnyModelProvider { content, label }| {
+                let view = self.app.new_view::<component_group::View>();
+                view.set_width(column_width);
+                view.set_entries(content);
+                view.set_header(label.as_str());
+                self.display_object.add_child(&view);
+                view
+            })
+            .collect_vec();
+
+        let mut columns = vec![vec![], vec![], vec![]];
+        let mut heights = [-COLUMN_GAP; 3]; // We need to subtract one COLUMN_GAP as we only need
+                                            // (n-1) gaps, but through iteration below we add one
+                                            // gap per item.
+
+        for (ix, entry) in content.iter().enumerate() {
+            let column_index = ix % 3;
+            columns[column_index].push(entry);
+            heights[column_index] += entry.size.value().y + COLUMN_GAP;
+        }
+        let height: f32 = heights.into_iter().map(OrderedFloat).max().unwrap().into();
+
+
+        let mut entry_ix = 0;
+        for (ix, column) in columns.iter().enumerate() {
+            let pos_x = (column_width + COLUMN_GAP) * (ix as f32 + 0.5);
+            let mut pos_y = -height;
+            for entry in column {
+                let entry_height = entry.size.value().y;
+                entry.set_position_y(pos_y + entry_height / 2.0);
+                entry.set_position_x(pos_x);
+
+                entry.set_color(get_base_color_for_index(entry_ix));
+                entry_ix += 1;
+
+                pos_y += entry_height;
+                pos_y += COLUMN_GAP;
+            }
+        }
+
+        *self.content.borrow_mut() = content;
+        let height: f32 = heights.into_iter().map(OrderedFloat).max().unwrap().into();
+        let width = self.size.get().x;
+        self.size.set(Vector2::new(width, height));
+        self.size.get()
+    }
+
+    pub fn set_layers(&self, layers: &Layers, scroll_layer: &display::scene::Layer) {
```

Comment:
no docs. If linter doesnt tell about it , it means it is wrongly configured. Could you please check it and write me a reply why linter did not raise warning here?